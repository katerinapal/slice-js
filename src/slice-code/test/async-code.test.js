import {runAllCombosTests, comboOfBools} from './helpers/utils'

/*
test.only(
  'orAndOr(false, false, true, true)',
  require('./helpers/utils').snapSlice(
    require.resolve('./fixtures/logical-expressions'),
    ({orAndOr}) => {
      return [orAndOr(false, false, true, true)]
    },
  ),
)
/* */

runAllCombosTests({
  filename: require.resolve('./fixtures/async-code'),
  methods: [
    {
      methodName: 'callPromise',
      possibleArguments: comboOfBools(1),
    },
    {
      methodName: 'callback',
      possibleArguments: comboOfBools(1),
    },
  ],
})
