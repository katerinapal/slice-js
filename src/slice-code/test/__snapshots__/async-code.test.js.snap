// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`callPromise(false) && callPromise(true) 1`] = `
"export {callPromise}

function promise(pass) {
  return new Promise((resolve, reject) => {
    if (pass) {
      resolve('pass')
    } else {
      reject('!pass')
    }
  })
}

function callPromise(pass) {
  return promise(pass).catch(rejection => rejection)
}
"
`;

exports[`callPromise(false) 1`] = `
"export {callPromise}

function promise() {
  return new Promise((resolve, reject) => {
    reject('!pass')
  })
}

function callPromise(pass) {
  return promise(pass).catch(rejection => rejection)
}
"
`;

exports[`callPromise(true) 1`] = `
"export {callPromise}

function promise() {
  return new Promise(resolve => {
    resolve('pass')
  })
}

function callPromise(pass) {
  return promise(pass).catch()
}
"
`;

exports[`callback(false) && callback(true) 1`] = `
"export {callPromise, callback}

function promise(pass) {
  return new Promise((resolve, reject) => {
    if (pass) {
      resolve('pass')
    } else {
      reject('!pass')
    }
  })
}

function callback(pass, cb) {
  return callPromise(pass).then(cb, cb)
}

function callPromise(pass) {
  return promise(pass).catch(rejection => rejection)
}
"
`;

exports[`callback(false) 1`] = `
"export {callPromise, callback}

function promise() {
  return new Promise((resolve, reject) => {
    reject('!pass')
  })
}

function callback(pass, cb) {
  return callPromise(pass).then(cb, cb)
}

function callPromise(pass) {
  return promise(pass).catch(rejection => rejection)
}
"
`;

exports[`callback(true) 1`] = `
"export {callPromise, callback}

function promise() {
  return new Promise(resolve => {
    resolve('pass')
  })
}

function callback(pass, cb) {
  return callPromise(pass).then(cb, cb)
}

function callPromise(pass) {
  return promise(pass).catch()
}
"
`;
